service/test/amqp_simulated.js

```diff
@@ -47,6 +47,26 @@ SimulatedAmqp.prototype.send = function send(deviceId, message, done) {
   }
 };
 
+SimulatedAmqp.prototype.sendToModule = function send(deviceId, moduleId, inputName, message, done) {
+  if (done) {
+    if (deviceId.search(/^no-device/) !== -1) {
```

Comment:
>/^no-device/ [](start = 24, length = 12)

not sure i'm liking this convention. would rather use stubs.

Reply:
egads!  I was just copy/paste coding.  I just have read deeper.  

---
In reply to: [180601661](https://github.com/Azure/azure-iot-sdk-node/pull/271#discussion_r180601661) [](ancestors = 180601661)

Reply:
actually, the test code works against this simulated transport and the 2 real transports.  If we use stubs in the test, it cripples the real transports.  I don't want to add "if simulated, add stub" to the test because that's just as ugly.

---
In reply to: [181218140](https://github.com/Azure/azure-iot-sdk-node/pull/271#discussion_r181218140) [](ancestors = 181218140,180601661)

Reply:
fair enough.

---
In reply to: [181473321](https://github.com/Azure/azure-iot-sdk-node/pull/271#discussion_r181473321) [](ancestors = 181473321,181218140,180601661)